OOP/oop_home_task.py

```diff
@@ -0,0 +1,475 @@
+            if symbol.isalnum():
+                symbols_count += 1
+        return symbols_count
+
+    def publish_article(self):
+        """
+        Blank method for publishing of Article that will be overwritten
+        @return: nothing
+        """
+        pass
+
+
+class ObjectsCreator:
+    """
+    Class for objects creation
+    """
+    @staticmethod
+    def create_news(city, publication_date, publication_text):
+        """
+        Creates a new news object with specified set of parameters
+        @param city: city where something happened
+        @param publication_date: date of publication
+        @param publication_text: text of publication
+        @return: created news object
+        """
+        news = News(city, publication_date, publication_text)
+        return news
+
+    @staticmethod
+    def create_advertisement(publication_date, expiration_date, publication_text):
+        """
+        Creates a new advertisement object with specified set of parameters
+        @param publication_date: date of publication
+        @param expiration_date: date when adv will expire
+        @param publication_text: text of publication
+        @return: created advertisement object
+        """
+        advertisement = Advertisement(publication_date, expiration_date, publication_text)
+        return advertisement
+
+    @staticmethod
+    def create_article(publication_text, author):
+        """
+        Creates a new article object with specified set of parameters
+        @param publication_text: text of publication
+        @param author: author of article
+        @return: created article object
+        """
+        article = Article(publication_text, author)
+        return article
+
+
+class FileWriter:
+    """
+    Class for writing to text file
+    """
+    @staticmethod
+    def open_file(file_name):
+        """
+        Open file for writing
+        @param file_name: file name
+        @return: opened file object
+        """
+        publications_file = open(f'{file_name}', 'a')
+        return publications_file
+
+    @staticmethod
+    def write_heading(publications_file, heading):
+        """
+        Writes heading of publication to file
+        @param publications_file: opened file object
+        @param heading: heading of publication
+        """
+        publications_file.write(f'{heading}\n')
+
+    @staticmethod
+    def write_publication_text(publications_file, text):
+        """
+        Writes text of publication to file
+        @param publications_file: opened file object
+        @param text: text of publication
+        """
+        publications_file.write(f'{text}\n')
+
+    @staticmethod
+    def write_publication_date(publications_file, publication_date):
+        """
+        Writes date of publication to file
+        @param publications_file: opened file object
+        @param publication_date: date of publication
+        """
+        publications_file.write(f'{publication_date}\n')
```

Comment:
Do you really need 3 separate classes for this? They all just write something to text file, we can easily replace them to one 'write' method. 
!!! Moreover we don't even need this method at all, because what a sense to implement a method with only one action? Just make pubplications_file.write where it's needed , no need use special methods for this.

Reply:
Removed classes - ObjectsCreator and FileWriter.